### PR TITLE
magento/merchdocs#1117: Expand catalog and products viewed compared topics

### PR DESCRIPTION
**Pull request by EvheniiaKolkhiieva**
## Purpose of this pull request

Fix pull request fixes https://github.com/magento/merchdocs/issues/1117

## Magento release version

- [x] 2.4.x

   Specify a patch release number, if applicable:

- [ ] 2.3.x

   Specify a patch release number, if applicable:

## Product editions

Is this update specific to a product edition?

- [x] Magento Open Source only
- [x] Magento Commerce only

Is this update specific to an installed feature extension

- [x] B2B extension
- [x] Other feature set, please specify:

## Affected documentation pages

- https://docs.magento.com/user-guide/v2.3/marketing/products-viewed-compared.html
- https://docs.magento.com/user-guide/configuration/catalog/catalog.html

## Links to Magento source code or PRs

- https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Catalog/Model/Product/ProductFrontendAction/Synchronizer.php

### CODE REVIEW

File: src/marketing/products-viewed-compared.md [hunk 1]
```diff
@@ -12,13 +12,19 @@ The Recently Viewed and Recently Compared blocks usually appear in the right sid
 
 1. Expand ![Expansion selector]({% link images/images/btn-expand.png %}) the **Recently Viewed/Compared Products** section and do the following:
 
-    ![Catalog configuration - recently viewed/compared products]({% link images/images/config-catalog-catalog-recently-viewed-compared-products.png %}){: .zoom}
+    ![Catalog configuration - recently viewed/compared products]({% link images/images/config-catalog-catalog-recently-viewed-compared-products1.png %}){: .zoom}
```
Comment (dobooth) [inline src/marketing/products-viewed-compared.md:15]: Why are there two images in this PR? If you are just updating an existing image, delete the original and replace it.

**Review by dobooth — CHANGES_REQUESTED** [review]

File: src/marketing/products-viewed-compared.md (hunk 1 above)
Comment (EvheniiaKolkhiieva) [reply src/marketing/products-viewed-compared.md]: Hi @dobooth, thank you for reviewing!
I have deleted both images and replaced them with a new one.
Please, check it :)

File: src/marketing/products-viewed-compared.md [hunk 2]
```diff
@@ -12,13 +12,19 @@ The Recently Viewed and Recently Compared blocks usually appear in the right sid
...
 
+   - Set **Synchronize widget products with backend storage** to synchronize product widget information such as product id with your current product storage availability in database. In order to reuse this info on different devices.
+
    - Set **Show for Current** to the website, store, or store view where the configuration applies.
 
    - In the **Default Recently Viewed Products Count** field, enter the number of recently viewed products to appear in the list.
 
    - In the **Default Recently Compared Products Count**, enter the number of recently compared products to appear in the list.
 
+   - In the **Lifetime of products in Recently Viewed Widget** enter how long the products viewed will be shown in the recently viewed list.
```
Comment (leslietilling) [inline src/marketing/products-viewed-compared.md:26]: The default value for these fields is `1000`... what does that equal? 1000 days, hours, something else?

Comment (EvheniiaKolkhiieva) [reply src/marketing/products-viewed-compared.md:26]: Hi @leslietilling, thank you for the great question!

According to the code, the default value for widgets is 1000 seconds
**Source code:**
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Catalog/Model/FrontendStorageConfigurationInterface.php#L20
**Regarding the range:**
There is no limit on the maximum amount of time, but at the same time, there is validation - only numbers and no less than zero.

`],
        ‘validate-greater-than-zero’: [
            function (v) {
                if ($.mage.isEmptyNoTrim(v)) {
                    return true;
                }
                v = $.mage.parseNumber(v);
                return !isNaN(v) && v > 0;
 },
            $.mage.__(‘Please enter a number greater than 0 in this field.’)
        ]`
**Source code:**
https://github.com/magento/magento2/blob/2.4/lib/web/mage/validation.js#L823

For more transparency for customers, I'll add clarification to the files and let you know. :)

Comment (EvheniiaKolkhiieva) [reply src/marketing/products-viewed-compared.md]: @leslietilling,
I have implemented some edits for src/marketing/products-viewed-compared.md  file. Could you please review and approve?

Thanks!

File: src/marketing/products-viewed-compared.md [hunk 3]
```diff
@@ -12,13 +12,19 @@ The Recently Viewed and Recently Compared blocks usually appear in the right sid
 
 1. Expand ![Expansion selector]({% link images/images/btn-expand.png %}) the **Recently Viewed/Compared Products** section and do the following:
 
-    ![Catalog configuration - recently viewed/compared products]({% link images/images/config-catalog-catalog-recently-viewed-compared-products.png %}){: .zoom}
+    ![Catalog configuration - recently viewed/compared products]({% link images/images/config-catalog-catalog-recently-viewed-and-compared-products.png %}){: .zoom}
     [_Recently Viewed/Compared Products_]({% link configuration/catalog/catalog.md %})
 
+   - Set **Synchronize widget products with backend storage** to synchronize product widget information such as product id with your current product storage availability in database. In order to reuse this info on different devices.
```
Comment (leslietilling) [inline src/marketing/products-viewed-compared.md:18]:
```suggestion
   - Set **Synchronize widget products with backend storage** to synchronize product widget information, such as product ID, with your current product storage availability in the database and reuse this information on different devices.
```

File: src/marketing/products-viewed-compared.md [hunk 4]
```diff
@@ -12,13 +12,19 @@ The Recently Viewed and Recently Compared blocks usually appear in the right sid
...
+
    - Set **Show for Current** to the website, store, or store view where the configuration applies.
 
    - In the **Default Recently Viewed Products Count** field, enter the number of recently viewed products to appear in the list.
 
    - In the **Default Recently Compared Products Count**, enter the number of recently compared products to appear in the list.
 
+   - In the **Lifetime of products in Recently Viewed Widget** set any time range in seconds, greater than zero. This field shows, how long the products viewed will be shown in the recently viewed list.
```
Comment (leslietilling) [inline src/marketing/products-viewed-compared.md:26]:
```suggestion
   - For **Lifetime of products in Recently Viewed Widget**, enter any time range in seconds, greater than zero. This setting determines how long the viewed products will be shown in the recently viewed list.
```

File: src/marketing/products-viewed-compared.md [hunk 5]
```diff
@@ -12,13 +12,19 @@ The Recently Viewed and Recently Compared blocks usually appear in the right sid
...
+
+   - In the **Lifetime of products in Recently Compared Widget** set any time range in seconds, greater than zero. This field shows, how long the products compared will be shown in the recently compared list.
```
Comment (leslietilling) [inline src/marketing/products-viewed-compared.md:28]:
```suggestion
   - For **Lifetime of products in Recently Compared Widget**, enter any time range in seconds, greater than zero. This setting determines how long the compared products will be shown in the recently compared list.
```

File: src/marketing/products-viewed-compared.md [hunk 6]
```diff
@@ -12,13 +12,19 @@ The Recently Viewed and Recently Compared blocks usually appear in the right sid
...
+    ![Catalog configuration - recently viewed/compared products]({% link images/images/config-catalog-catalog-recently-viewed-and-compared-products.png %}){: .zoom}
     [_Recently Viewed/Compared Products_]({% link configuration/catalog/catalog.md %})
 
+   - Set **Synchronize widget products with backend storage** to synchronize product widget information such as product id with your current product storage availability in database. In order to reuse this info on different devices.
+
    - Set **Show for Current** to the website, store, or store view where the configuration applies.
 
    - In the **Default Recently Viewed Products Count** field, enter the number of recently viewed products to appear in the list.
 
    - In the **Default Recently Compared Products Count**, enter the number of recently compared products to appear in the list.
```
Comment (leslietilling) [inline src/marketing/products-viewed-compared.md:24]: I n=know you did not update this one, but let's fix while we are here.

```suggestion
   - For **Default Recently Compared Products Count**, enter the number of recently compared products to appear in the list.
```

File: src/configuration/catalog/catalog.md [hunk 7]
```diff
@@ -93,9 +93,12 @@ Stores > Settings > [Configuration]({% link stores/configuration.md %}) > [Cata
 
 |Field|[Scope]({% link configuration/scope.md %})|Description|
 |--- |--- |--- |
+|Synchronize widget products with backend storage|Global|Determines the sync of product widget information, such as product id, with database. Allows the reuse of info on other devices.|
```
Comment (leslietilling) [inline src/configuration/catalog/catalog.md:96]:
```suggestion
|Synchronize widget products with backend storage|Global|Determines the sync of product widget information, such as product ID, with the database. This allows the reuse of information on other devices.|
```

File: src/configuration/catalog/catalog.md [hunk 8]
```diff
@@ -93,9 +93,12 @@ Stores > Settings > [Configuration]({% link stores/configuration.md %}) > [Cata
...
 |Show for Current|Website|Limits the products shown to the current website. Options: Website / Store / Store View|
 |Default Recently Viewed Products Count|Store View|Determines the maximum number of recently viewed products that appear in the list.|
 |Default Recently Compared Products Count|Store View|Determines the maximum number of recently compared products that appear in the list.|
+|Lifetime of products in Recently Viewed Widget|Global|Determines how long the products will be shown in the recently viewed list.|
```
Comment (leslietilling) [inline src/configuration/catalog/catalog.md:100]:
```suggestion
|Lifetime of products in Recently Viewed Widget|Global|Determines how long, in seconds, viewed products will be shown in the recently viewed list.|
```

File: src/configuration/catalog/catalog.md [hunk 9]
```diff
@@ -93,9 +93,12 @@ Stores > Settings > [Configuration]({% link stores/configuration.md %}) > [Cata
...
+|Lifetime of products in Recently Compared Widget|Global|Determines how long the products will be shown in the recently compared list.|
```
Comment (leslietilling) [inline src/configuration/catalog/catalog.md:101]:
```suggestion
|Lifetime of products in Recently Compared Widget|Global|Determines how long, in seconds, compared products will be shown in the recently compared list.|
```

**Review by leslietilling — CHANGES_REQUESTED** [review]
Just some editorial updates and will be good to go.

**Comment by EvheniiaKolkhiieva** [comment]
Hi @leslietilling.

I have implemented your updates, please approve.

Thank you!

File: src/configuration/catalog/catalog.md (hunk 9 above)
Comment (EvheniiaKolkhiieva) [reply src/configuration/catalog/catalog.md]: Done :)

File: src/configuration/catalog/catalog.md (hunk 8 above)
Comment (EvheniiaKolkhiieva) [reply src/configuration/catalog/catalog.md]: Done :)

File: src/marketing/products-viewed-compared.md (hunk 6 above)
Comment (EvheniiaKolkhiieva) [reply src/marketing/products-viewed-compared.md]: Done :)

File: src/marketing/products-viewed-compared.md (hunk 5 above)
Comment (EvheniiaKolkhiieva) [reply src/marketing/products-viewed-compared.md]: Done :)

File: src/marketing/products-viewed-compared.md (hunk 4 above)
Comment (EvheniiaKolkhiieva) [reply src/marketing/products-viewed-compared.md]: Done :)

File: src/marketing/products-viewed-compared.md (hunk 3 above)
Comment (EvheniiaKolkhiieva) [reply src/marketing/products-viewed-compared.md]: Done :)

**Review by leslietilling — APPROVED** [review]

**Comment by leslietilling** [comment]
@dobooth  You requested changes, so this needs your approval as well

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
@magento import code to magento-devdocs/merchdocs

**Comment by magento-engcom-team** [comment]
@dobooth the branch with code successfully imported into`magento-devdocs/merchdocs` repository. Branch name: `imported-magento-merchdocs-1163`.

**Pull request merged**